File: tikito-ui/src/app/dto/server-response.ts
export class ServerResponse {
    static ResourceNotFoundException = 'ResourceNotFoundException';
    static ResourceAlreadyExistsException = 'ResourceAlreadyExistsException';
    static InvalidCredentialsException = 'InvalidCredentialsException';
    static GroupNotEmptyException = 'GroupNotEmptyException';
    static InvalidContactException = 'InvalidContactException';

    error: string;
    message: string;
    data: any;
}

File: tikito-backend/src/main/java/org/tikito/controller/ServerResponse.java
package org.tikito.controller;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class ServerResponse<T> {
    private String error;
    private String message;
    private T data;

    public ServerResponse(final String error, final T data) {
        this.error = error;
        this.data = data;
    }

    public ServerResponse(final Exception e) {
        this.error = e.getClass().getSimpleName();
        this.message = e.getMessage();
    }
}
